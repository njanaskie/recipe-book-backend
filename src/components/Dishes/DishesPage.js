import React, { useReducer } from 'react'
import dishesReducer from '../../reducers/dishes'
import DishesContext from '../../../context/dishes-context'
import DishesList from '../Dishes/DishesList'
import FiltersContext from '../../../context/filters-context'
import filtersReducer from '../../reducers/filters'

const DishesPage = () => {
    const [dishes, dishDispatch] = useReducer(dishesReducer, [])
    const [filters, filtersDispatch] = useReducer(filtersReducer)

    return (
        <FiltersContext.Provider value={{ filters, filtersDispatch }}>
            <DishesContext.Provider value={{ dishes, dishDispatch }}>
                <h3>Dishes Page</h3>
                <DishesList />
            </DishesContext.Provider>
        </FiltersContext.Provider>
    )

};

export default DishesPage;