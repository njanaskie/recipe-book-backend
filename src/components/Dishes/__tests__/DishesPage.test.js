import React from 'react'
import { shallow } from 'enzyme'
import DishesPage from '../DishesPage'

test('should render DishesPage correctly',() => {
    const wrapper = shallow(<DishesPage />)
    expect(wrapper).toMatchSnapshot()
})