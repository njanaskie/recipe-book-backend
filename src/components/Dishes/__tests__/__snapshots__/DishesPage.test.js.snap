// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DishesPage correctly 1`] = `
<ContextProvider
  value={
    Object {
      "dishDispatch": [Function],
      "dishes": Array [],
    }
  }
>
  <h3>
    Dishes Page
  </h3>
  <DishesList />
</ContextProvider>
`;
